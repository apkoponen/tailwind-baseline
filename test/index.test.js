const plugin = require("../src/index");
const invokePlugin = require("./_utils/invokePlugin");

test("should work with multiple fontSizes and lineHeights", () => {
  const config = {
    theme: {
      spacing: {
        "1": "0.25rem"
      },
      fontSize: {
        sm: "0.875rem",
        base: "1rem",
        "5xl": "3rem"
      },
      lineHeight: {
        none: "1",
        snug: "1.375",
        "3": ".75rem",
        "4": "1rem"
      },
      fontFamily: {
        sans: ["Inter var"]
      },
      baseline: {
        fonts: {
          sans: {
            descenderHeightScale: 0.136,
            capHeight: 0.723
          }
        }
      }
    },
    variants: {}
  };

  const { utilities } = invokePlugin(plugin.handler, config);
  const utilityObject = utilities[0][0];
  const utilityKeys = Object.keys(utilityObject);

  const textSmLeadningNone = utilityObject[utilityKeys[0]];
  expect(textSmLeadningNone.fontSize).toEqual("0.875rem");
  expect(textSmLeadningNone.lineHeight).toEqual("1rem");
  expect(textSmLeadningNone.transform).toEqual(
    "translateY(0.2074285714285714em)"
  );
  expect(textSmLeadningNone["&:before"].marginTop).toEqual("-0.5rem");

  expect(utilityObject).toMatchSnapshot();
});

test("should generate styles for all configured fonts", () => {
  const config = {
    theme: {
      spacing: {
        "1": "0.25rem"
      },
      fontSize: {
        base: "1rem"
      },
      lineHeight: {
        none: "1"
      },
      fontFamily: {
        sans: ["Inter var"],
        serif: ["Merriweather"],
        mono: ["Roboto Mono"]
      },
      baseline: {
        fonts: {
          sans: {
            descenderHeightScale: 0.136,
            capHeight: 0.723
          },
          serif: {
            descenderHeightScale: 0.144,
            capHeight: 0.7462
          },
          mono: {
            descenderHeightScale: 0.111,
            capHeight: 0.7077
          }
        }
      }
    },
    variants: {}
  };

  const { utilities } = invokePlugin(plugin.handler, config);
  const classNames = Object.keys(utilities[0][0]);

  expect(classNames).toHaveLength(5);
  expect(classNames[0]).toContain(".font-sans");
  expect(classNames[1]).toContain(".font-serif");
  expect(classNames[2]).toContain(".font-mono");
});
