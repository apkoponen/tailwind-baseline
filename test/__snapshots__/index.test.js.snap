// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should work with multiple fontSizes and lineHeights 1`] = `
Object {
  ".baseline-debug-line": Object {
    "background": "linear-gradient(rgba(255, 0, 0, 0.15), rgba(255, 0, 0, 0.15) 1px, transparent 1px)",
    "backgroundSize": "1px 0.25rem",
  },
  ".baseline-debug-stripe": Object {
    "background": "repeating-linear-gradient(0, #68d391,#68d391 0.25rem, #c6f6d5 0.25rem, #c6f6d5 0.5rem)",
  },
  ".font-sans.text-5xl.leading-3.baseline, .font-sans .text-5xl.leading-3.baseline, .font-sans .text-5xl .leading-3.baseline ,.text-5xl .font-sans.leading-3.baseline, .text-5xl .font-sans .leading-3.baseline": Object {
    "&:before": Object {
      "content": "''",
      "display": "block",
      "height": 0,
      "marginTop": "-0.25rem",
    },
    "fontSize": "3rem",
    "lineHeight": "0.75rem",
    "paddingTop": "0.25rem",
    "transform": "translateY(-0.239em)",
  },
  ".font-sans.text-5xl.leading-4.baseline, .font-sans .text-5xl.leading-4.baseline, .font-sans .text-5xl .leading-4.baseline ,.text-5xl .font-sans.leading-4.baseline, .text-5xl .font-sans .leading-4.baseline": Object {
    "&:before": Object {
      "content": "''",
      "display": "block",
      "height": 0,
      "marginTop": "-0.25rem",
    },
    "fontSize": "3rem",
    "lineHeight": "1rem",
    "paddingTop": "0.25rem",
    "transform": "translateY(-0.19733333333333336em)",
  },
  ".font-sans.text-5xl.leading-none.baseline, .font-sans .text-5xl.leading-none.baseline, .font-sans .text-5xl .leading-none.baseline ,.text-5xl .font-sans.leading-none.baseline, .text-5xl .font-sans .leading-none.baseline": Object {
    "&:before": Object {
      "content": "''",
      "display": "block",
      "height": 0,
      "marginTop": "-1rem",
    },
    "fontSize": "3rem",
    "lineHeight": "3rem",
    "paddingTop": "0.25rem",
    "transform": "translateY(0.136em)",
  },
  ".font-sans.text-5xl.leading-snug.baseline, .font-sans .text-5xl.leading-snug.baseline, .font-sans .text-5xl .leading-snug.baseline ,.text-5xl .font-sans.leading-snug.baseline, .text-5xl .font-sans .leading-snug.baseline": Object {
    "&:before": Object {
      "content": "''",
      "display": "block",
      "height": 0,
      "marginTop": "-2.25rem",
    },
    "fontSize": "3rem",
    "lineHeight": "4.25rem",
    "paddingTop": "0.25rem",
    "transform": "translateY(0.3443333333333334em)",
  },
  ".font-sans.text-base.leading-3.baseline, .font-sans .text-base.leading-3.baseline, .font-sans .text-base .leading-3.baseline ,.text-base .font-sans.leading-3.baseline, .text-base .font-sans .leading-3.baseline": Object {
    "&:before": Object {
      "content": "''",
      "display": "block",
      "height": 0,
      "marginTop": "-0.25rem",
    },
    "fontSize": "1rem",
    "lineHeight": "0.75rem",
    "paddingTop": "0.25rem",
    "transform": "translateY(0.01100000000000001em)",
  },
  ".font-sans.text-base.leading-4.baseline, .font-sans .text-base.leading-4.baseline, .font-sans .text-base .leading-4.baseline ,.text-base .font-sans.leading-4.baseline, .text-base .font-sans .leading-4.baseline": Object {
    "&:before": Object {
      "content": "''",
      "display": "block",
      "height": 0,
      "marginTop": "-0.5rem",
    },
    "fontSize": "1rem",
    "lineHeight": "1rem",
    "paddingTop": "0.25rem",
    "transform": "translateY(0.136em)",
  },
  ".font-sans.text-base.leading-none.baseline, .font-sans .text-base.leading-none.baseline, .font-sans .text-base .leading-none.baseline ,.text-base .font-sans.leading-none.baseline, .text-base .font-sans .leading-none.baseline": Object {
    "&:before": Object {
      "content": "''",
      "display": "block",
      "height": 0,
      "marginTop": "-0.5rem",
    },
    "fontSize": "1rem",
    "lineHeight": "1rem",
    "paddingTop": "0.25rem",
    "transform": "translateY(0.136em)",
  },
  ".font-sans.text-base.leading-snug.baseline, .font-sans .text-base.leading-snug.baseline, .font-sans .text-base .leading-snug.baseline ,.text-base .font-sans.leading-snug.baseline, .text-base .font-sans .leading-snug.baseline": Object {
    "&:before": Object {
      "content": "''",
      "display": "block",
      "height": 0,
      "marginTop": "-1rem",
    },
    "fontSize": "1rem",
    "lineHeight": "1.5rem",
    "paddingTop": "0.25rem",
    "transform": "translateY(0.386em)",
  },
  ".font-sans.text-sm.leading-3.baseline, .font-sans .text-sm.leading-3.baseline, .font-sans .text-sm .leading-3.baseline ,.text-sm .font-sans.leading-3.baseline, .text-sm .font-sans .leading-3.baseline": Object {
    "&:before": Object {
      "content": "''",
      "display": "block",
      "height": 0,
      "marginTop": "-0.25rem",
    },
    "fontSize": "0.875rem",
    "lineHeight": "0.75rem",
    "paddingTop": "0.25rem",
    "transform": "translateY(0.06457142857142856em)",
  },
  ".font-sans.text-sm.leading-4.baseline, .font-sans .text-sm.leading-4.baseline, .font-sans .text-sm .leading-4.baseline ,.text-sm .font-sans.leading-4.baseline, .text-sm .font-sans .leading-4.baseline": Object {
    "&:before": Object {
      "content": "''",
      "display": "block",
      "height": 0,
      "marginTop": "-0.5rem",
    },
    "fontSize": "0.875rem",
    "lineHeight": "1rem",
    "paddingTop": "0.25rem",
    "transform": "translateY(0.2074285714285714em)",
  },
  ".font-sans.text-sm.leading-none.baseline, .font-sans .text-sm.leading-none.baseline, .font-sans .text-sm .leading-none.baseline ,.text-sm .font-sans.leading-none.baseline, .text-sm .font-sans .leading-none.baseline": Object {
    "&:before": Object {
      "content": "''",
      "display": "block",
      "height": 0,
      "marginTop": "-0.5rem",
    },
    "fontSize": "0.875rem",
    "lineHeight": "1rem",
    "paddingTop": "0.25rem",
    "transform": "translateY(0.2074285714285714em)",
  },
  ".font-sans.text-sm.leading-snug.baseline, .font-sans .text-sm.leading-snug.baseline, .font-sans .text-sm .leading-snug.baseline ,.text-sm .font-sans.leading-snug.baseline, .text-sm .font-sans .leading-snug.baseline": Object {
    "&:before": Object {
      "content": "''",
      "display": "block",
      "height": 0,
      "marginTop": "-0.75rem",
    },
    "fontSize": "0.875rem",
    "lineHeight": "1.25rem",
    "paddingTop": "0.25rem",
    "transform": "translateY(0.3502857142857143em)",
  },
}
`;
